common/flatpak-dir.c

```diff
@@ -3934,18 +3921,52 @@ _flatpak_dir_find_new_flatpakrepos (FlatpakDir *self, OstreeRepo *repo)
         {
           g_autofree char *remote_name = g_strndup (name, strlen (name) - strlen (SYSCONF_REMOTES_FILE_EXT));
 
-          if (remotes && g_strv_contains ((const char * const *)remotes, remote_name))
+          if (remotes && g_strv_contains (remotes, remote_name))
             continue;
 
-          if (applied_remotes && g_strv_contains ((const char * const *)applied_remotes, remote_name))
-            continue;
+          if (*flatpakrepos == NULL)
+            *flatpakrepos = g_hash_table_new_full (g_str_hash, g_str_equal, g_free, g_object_unref);
```

Comment:
I would move this to `_flatpak_dir_find_new_flatpakrepos`

Reply:
The problem there is that `_flatpak_dir_find_new_flatpakrepos` will then always return something that is not `NULL`, and the consumer is not expecting that. Right now there is a warranty that if it's non-null, it contains at least one repo. Is there a reason to change this?

Reply:
Good point, let's leave it as is.